src/registrar/views/application.py

```diff
@@ -329,6 +335,41 @@ def pending_applications(self):
         ]
         return DomainApplication.objects.filter(creator=self.request.user, status__in=check_statuses)
 
+    def db_check_for_unlocking_steps(self):
+        """Helper for get_context_data
+
+        Queries the DB for an application and returns a dict for unlocked steps."""
+        history_dict = {
```

Comment:
(Optional) Its fine given the context of the ticket, but I think this can be generalized a bit using a separate function and getattr (since your already storing a list of variable names). Then you could just run the bool logic on the returned boolean array.

For instance (bad example):
attr_list = [
"org_type",
etc
]

thing = some_function(application, atttr_List)
return [for thingy in thing if thingy]

def some_function(self, application, attr_list):
  bools = {}
  for attr in attr_list:
    bools["attr"] = True if getattr(application, attr, None)) else False
  return bools